fdbserver/include/fdbserver/StorageMetrics.actor.h

```diff
@@ -152,3 +157,76 @@ struct ByteSampleInfo {
 // Determines whether a key-value pair should be included in a byte sample
 // Also returns size information about the sample
 ByteSampleInfo isKeyValueInSample(KeyValueRef keyValue);
+
+class IStorageMetricsService {
+public:
+	StorageServerMetrics metrics;
+
+	// penalty used by loadBalance() to balance requests among service instances
+	virtual double getPenalty() { return 1; }
+
+	virtual bool isReadable(KeyRangeRef const& keys) { return true; }
```

Comment:
These functions can be marked `const`. Nit: IMO it would be better to make them pure virtual instead of only overriding in one of two child classes.

Reply:
I think for some child classes, like MGS and BlobMigrator, the `getPenalty()` and `isReadable()` just want to share the same simple implementations. So I prefer to have those simple implementations in the parent class.